### ISSUE TITLE
java.lang.IllegalArgumentException: Info not on store!

### ISSUE THREAD

**Issue by janbolat** [lingochamp/okdownload#34]
Hi!
Thank you for the library!
I couldn't understand couple things because it is not well documented. Could you please help me?

1) What is the difference between RemitStore and BreakpointStore. 
I tried to initialize OkDownload like following:
```
OkDownload.setSingletonInstance(OkDownload.Builder(appContext)
                                .downloadStore(BreakpointStoreOnSQLite(appContext).createRemitSelf())
                               // .downloadStore(BreakpointStoreOnSQLite(appContext)) // and like this
                                .build())
```
But data is not written to the database.
I have added implementation "com.liulishuo.okdownload:sqlite:1.0.1-SNAPSHOT" in build.gradle but couldn't use it. How can I add sqlite support?

2) How can I cancel a task (delete completely all information about task)? 
Right now I am doing like this:
```
tasks[task].cancel() 
OkDownload.with().breakpointStore().remove(task.id)
```
But I am getting the FATAL Exception:
`java.lang.IllegalArgumentException: Info not on store!`

Hope to hear from u soon.

**Comment by Jacksgong** [comment]
1. as the [wiki](https://github.com/lingochamp/okdownload/wiki) said: `If you import component library it would be enabled by default`, so after you implementation `sqlite` in `build.gradle` , it will be enabled by default. You also can ensure it through the code of `OkDownload.Builder#build()` or check log if you really want to finger out the problem. Although it's no need to `setSingletonInstance` again for valid `sqlite`, your code is also right, the `sqlite` would be enabled too as your code write, you also can through the log, debug the program.
2.  After you `cancel` the task, For cancel operation is more responsible, the sync data to store job is running asynchronously, but the data is removed from you. **This crash does not make sense,  it will be fixed on the 1.0.1-SNAPSHOT**.
3. RemitDatabase, what is used for optimizing the case of many small tasks raise many useless database operations:

![remit-database](https://user-images.githubusercontent.com/6176417/38881879-71ac5290-429b-11e8-9424-c86820603000.png)


> P.S. You receive such FATAL Exception from the `BreakpointStoreOnCache` is because the `sqlite` breakpoint-store also use `cache` breakpoint-store to speed up `read`.

**Comment by Jacksgong** [comment]
Now the `FATAL Exception` is fixed on the latest 1.0.1-SNAPSHOT.
